src/main/java/com/agnieszkapawska/flashcards/domain/facades/FlashcardFacade.java

```diff
@@ -33,4 +39,35 @@ public FlashcardSaveResponseDto saveFlashcard(FlashcardDto flashcardDto) {
             throw new EntityNotCreatedException("something went wrong");
         }
     }
+
+    public FlashcardSaveResponseDto updateFlashcard(FlashcardDto flashcardDto, Long id) {
+        Flashcard flashcardFound = flashcardService.findById(id);
+
+        CompareQuestionTagsSets tagsToUpdate = compareQuestionTagSets(flashcardFound.getQuestionTagsSet(), flashcardDto.getTagsList());
+        questionTagService.updateFlashcardSet(tagsToUpdate, flashcardFound);
+
+        Set<QuestionTag> questionTagsToAddSet = questionTagService.getQuestionTagsSet(tagsToUpdate.getTagsNamesToAdd());
+        Set<QuestionTag> questionTagsToRemoveSet = questionTagService.getQuestionTagsSet(tagsToUpdate.getTagsNamesToRemove());
```

Comment:
Now when you have those objects here you could consider moving logic from questionTagService.updateFlashcardSet here.

Reply:
I'm not sure if I understand what you meant but I changed it.